tests/test_symbolics.py

```diff
@@ -313,7 +313,7 @@ def test_cos_vs_cosf():
 
     # Doesn't make much sense, but it's legal
     c = dSymbol('c', dtype=np.int32)
-    assert ccode(cos(c)) == "cos(c)"
+    assert ccode(cos(c)) == "cosf(c)"
```

Comment:
so `ccode(Abs(c))` will be `absf(c)` which is incorrect it should be `abs(c)`.

Reply:
You are correct, there is a bug here with your specific example. However, I think I introduced it in my last PR and it is specific to `abs`/`fabsf`/`fabs`, which I will address.

But, in this test, why should the resultant code be `cos(c)` over `cosf(c)` when `c` is an integer type?

Reply:
Well `cos` is technically not defined for integers so it will convert to float or double depending on the function used so I guess the test is broken in itself and should use a function that is valid for integers